lib/core/repositories/auth/auth_repository.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:live_score/core/repositories/auth/auth_repository_interface.dart';
+
+class AuthRepository implements AuthRepositoryInterface {
+  final FirebaseAuth _auth;
+
+  AuthRepository({required FirebaseAuth auth}) : _auth = auth;
+
+  @override
+  Future<User?> signUpWithEmailAndPassword(
+      String email, String password) async {
+    try {
+      UserCredential credential = await _auth.createUserWithEmailAndPassword(
+        email: email,
+        password: password,
+      );
+      return credential.user;
+    } on FirebaseAuthException catch (e) {
+      throw FirebaseAuthException(code: e.code);
```

Comment:
I don't see a point to use try cache and throwing same error.

Reply:
Fixed!